ERCS/erc-7654.md

```diff
@@ -0,0 +1,182 @@
+---
+eip: 7654
+title: Request Method Types
+description: Use a set of request methods to indicate the type of action to take on the contract.
+author: Rickey (@HelloRickey)
+discussions-to: https://ethereum-magicians.org/t/erc-7654-request-method-types/19183
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-13
+---
+
+## Abstract
+
+This proposal standardizes a set of request and response communication standards between clients and smart contracts, using POST, GET, and PUT requests to create, read, and update the states of smart contracts. You can customize different request method names, request parameters and response values, and each request method will be mapped to a specific operation.
+
+## Motivation   
+
+In Web2.0, we often use HTTP methods to define how the client interacts with the server. We use the POST, GET, and PUT methods to perform different operations on the resources in the server. This standard effectively simplifies the interaction between the client and the server. Similar standards can also be used to wrap functions in smart contracts. When the client makes a request to the contract, it indicates the type and method of operation to be taken, and the corresponding method in the contract performs the operation and returns a response.
+
+Advantages of using this standard include:
+
+- Standardization: Provide a unified and standardized communication method for clients and contracts, so that all parties can conduct more standardized interactions.
+
+- Composability: It can customize request methods, request parameters and response values, and easily build complex applications with multiple contract request methods components.
+
+- Interoperability: Easily extend plugins or middleware in request methods to enable interoperability with onchain applications from all parties.
+
+- State Management: Create, retrieve, and modify smart contract states according to the application use cases of each party.
+
+- RESTful: It can adopt RESTful design principles to make it more predictable and make calls to all parties clearer and easier to understand.
```

Comment:
Very little of your Motivation is specific to HTTP-style verbs, and can equally be applied to _any_ standard interface for contracts. You should be trying to convince the reader that contracts should implement specifically `GET`/`POST`/etc., not why we need standards generally.